### DISCUSSION TITLE
macros.cfg from SR LGXlite

### DISCUSSION THREAD

**Discussion by CaryBab** [Guilouz/Klipper-Flsun-Speeder-Pad#124] [General]
In the START_PRINT section
   {% if printer.extruder.target != 0 %} seems to be incorrect.  
   With this if statement my temp always get set to 150
   Commenting out this IF statement set my temp get set correctly

In the RESUME section
   You have an M106 S255
   Which is annoying at least, but if you are using a filament that requires zero cooling it is print destroying.

In the PAUSE section
   You use a M109 S0 to turn off heating.
   I believe it should be M104 S0 and remove the M106 S0
   I understand you do this because you next turn off cooling.
  But if you are doing a filament change you don't really want the temperature to be 0 before you change the filament.
  And if you don't shut off cooling, you won't have to start it up in the RESUME section

**Comment by Guilouz** [comment]
This is wanted in the start print macro, it is to start heating the nozzle to 150 while the bed heats up.

I will rework pause/resume/m600 macro for fan speed.

Yes it's a mistake for M109 S0, i will fix it.

**Reply by CaryBab** [reply]
I wasn't saying not to heat up to 150.  I was saying "printer.extruder.target" is not correct.  I think it should be "printer.extruder.temperature"

**Reply by Guilouz** [reply]
No it's correct, it's target temp that it's checked.

**Comment by CaryBab** [comment]
Mine gets set to 150 even though I have my Cura start code defined like
your start code.

If I change it to "printer.extruder.temperature" it sets it to the correct
temperature.

Maybe it's just me.

Cary

**Reply by Guilouz** [reply]
Yes this is normal. Here is the process:

If the hotend is not preheated before print then it heats to 150 and waits for the bed to heat up before returning to its target temperature.

If the hotend is already preheated then it remains at its target temperature while the bed is heated.

**Comment by CaryBab** [comment]
Is there a reason why you wouldn't want the extruder to heat up to target
temperature everytime and eliminate the If statement entirely ?

**Reply by Guilouz** [reply]
Because the bed takes much longer to heat up and it is therefore to avoid heating the filament without extrusion and avoid filament flows.

**Comment by CaryBab** [comment]
Gotcha, The fact that you move the hotend off to the side helps a lot
also.  Thanks.
